src/AntiCSRF.php

```diff
@@ -548,4 +544,34 @@ protected static function noHTML(string $untrusted): string
     {
         return \htmlentities($untrusted, ENT_QUOTES, 'UTF-8');
     }
+
+    /**
+     * Wrapper to support different sources of remote IP addresses.
+     *
+     * @return string
+     */
+    protected function getRemoteIP(): string
+    {
+        if (isset($this->server["HTTP_CF_CONNECTING_IP"]) && filter_var($this->server["HTTP_CF_CONNECTING_IP"], FILTER_VALIDATE_IP))
+        {
+            return $this->server["HTTP_CF_CONNECTING_IP"];
```

Comment:
What if an attacker specifies a fake header here after discovering the server's real IP?

What if an attacker specifies this header for a non-CloudFlare IP?

Etc.

Reply:
Would you suggest validating against the REMOTE_ADDR to the CloudFlare IP ranges? https://www.cloudflare.com/ips/

Reply:
Yes, and this can be hard-coded. We can update it in the future if their list of IPs widens.